### ISSUE TITLE
problem with 2.0.0-rc.1 / angualr 5

### ISSUE THREAD

**Issue by JanEggers** [valor-software/ng2-tree#160]
when starting the app i get the following error after updating to 2.0.0-rc.1 and ng5:

Unexpected value 'TreeModule' imported by the module 'AppModule'. Please add a @NgModule annotation.

if i stay on ng4 i get another error on importing TreeModule saying expected version 3 but value is 4

<details>
 <summary>solved</summary>

If I change back to ng2-tree 2.0.0-alpha10 (but still update from ng 4.4.6 to 5.0.0)  i get an error :
Can't bind to 'ngOutletContext' since it isn't a known property of 'ng-template'. from ng:///NgxDatatableModule/DataTableHeaderCellComponent.html@20:6

(im also using ng2-table 1.3.2) lastest version of that library that still depends on angular 2.0.

so can you guys at valorsoft please coordinate your dependency upgrade patterns.

it was not ng2-table but @swimlane/ngx-datatable that could not handle ng5
</details>

**Comment by jszuslik** [comment]
@JanEggers You marked it as solved but what is the resolution? I am getting the error ' importing TreeModule saying expected version 3 but value is 4' How did you resolve that?

**Comment by JanEggers** [comment]
@jszuslik  no i just solved what was related to ng5 update, thats why i masked it from the issue description. you can click on solved to see the other issue i had. the issue with ngtree still remains otherwise i would have colsed this issue.

**Comment by SpiraMira** [comment]
I get this too after upgrading my project to ng5 to support 2.0.0-rc1.

**Comment by SpiraMira** [comment]
@jszuslik you probably get hat cause you are importing an ng5 module into an ng4 project. Got that too but now I get the OP's issue.

**Comment by SpiraMira** [comment]
Just an update : my work around is to import the src tree instead of npm install. Seems like a packaging issue.

**Comment by JanEggers** [comment]
@SpiraMira nice at least a workaround

**Comment by rychkog** [comment]
@JanEggers @SpiraMira I'll try to reproduce and fix. So this happens only when we use ng-tree.rc-1 in a ng4 project, right?

**Comment by JanEggers** [comment]
i you stay on ng4 it gives metadata error. if you update to ng5 it says missing ngmodule annotation,

**Comment by rychkog** [comment]
@JanEggers Got it

**Comment by rychkog** [comment]
@JanEggers What typescript version do you have?

**Comment by JanEggers** [comment]
2.3 before updateing to ng5 and 2.4.2 after as thats the minimum required version of ng5

**Comment by amir2260** [comment]
i have same problem after update to angular5, how to fix ?

**Comment by jszuslik** [comment]
Update your packages one by one until it works. I did that and got it working just fine.

**Comment by TMaster** [comment]
What packages did you update? the dependencies? anuglar?

**Comment by michaljabi** [comment]
I had similar problems...

1. Updated Angular to `5.0.1` form `4.4.6` and `ng2-tree` as well to version `2.0.0-rc.1`
Then got a message:

```Unexpected value 'TreeModule' imported by the module 'FolderViewModule'. Please add a @NgModule annotation.```
It is a run-time error

2. I decided to downgrade ng-tree to version `2.0.0-alpha.8` but I got a message:

```ERROR in Error: Metadata version mismatch for module (...)/node_modules/ng2-tree/index.d.ts, found version 4, expected 3, resolving symbol (...)```
It is a compiler error

3. I decided to downgrade all dependencies (so go back in time and do nothing) but still got a message:

```ERROR in Error: Metadata version mismatch...``` again !

Problem was that I had always done it by deleting all node_modules and than updating package.json file (dependency by dependency) an then running `npm install`

When I read suggestion of 
@SpiraMira that it 
> Seems like a packaging issue.  

I figured that maby runing command `npm install ng2-tree@2.0.0-alpha.8` separately will solve my issue

**And that solve the problem.**

### TL;DR

You can update angular to **>5.0** but after runnig `npm install` you need to run separately command  
 `npm install ng2-tree@2.0.0-alpha.8`  
 -> version depends on what you got before, in my case problem still exist on `2.0.0-rc.1` but I was able to "upgrade" to `2.0.0-alpha.10` 

so finally after updating angular libs to `5.0.1` exactly
I ran `npm install 2.0.0-alpha.10` command additionally

Looks like `2.0.0-rc.1` version need to be taken care of - to repair this issue.

**Comment by SpiraMira** [comment]
@rychkog Hi - any updates on rc-1?  My "work-around" (copying the src) is obviously not optimal but that works fine with ng ^5.0.1 and related dependencies.  With typescript ~2.4.2 and or VS code typescript  2.4.2 or 2.6.1 (latest)

Also looks like we're not the only ones : see angular/material#8229 and swimlane/ngx-datatable#1104. 

By the way, what is the recommended way of building your lib locally? I would like to npm link that development build to my local project.

Regards

**Comment by rychkog** [comment]
@SpiraMira @JanEggers 
Try to use ng2-tree@2.0.0-rc.2
typescript 2.4.2

**Comment by JanEggers** [comment]
rc2 looks good thx

**Comment by TMaster** [comment]
Still happens to me, Must I be in Angular 5?
